File: algorithms/Path Sum/Path Sum.java
class Solution {
    public boolean hasPathSum(TreeNode root, int targetSum) {
       if(root == null) return false;
      
      if(root.left == null && root.right == null) return root.val == targetSum;
      
      return hasPathSum(root.right, targetSum - root.val) || hasPathSum(root.left, targetSum - root.val);
    }
}

File: scripts/algorithms/P/Path Sum/Path Sum.cpp
class Solution {
public:
    bool ans=false;
      int sum=0;
    void recur(TreeNode* root, int target){
        if(root==NULL)return;
        
            sum+=root->val;
            recur(root->left,target);
            recur(root->right,target);
           if(root->left==NULL && root->right==NULL&&sum == target){ // !!Check only if it is a leaf node....
             ans = true;
             return;
            }
            sum-=root->val; //backtracking
            return;
        
    }
    bool hasPathSum(TreeNode* root, int targetSum) {
        recur(root,targetSum);
        return ans;
    }
};
